Mathlib/Data/FunLike/Basic.lean

```diff
@@ -141,7 +141,7 @@ variable {F α β} [i : FunLike F α β]
 
 -- Give this a priority between `coe_fn_trans` and the default priority
 -- `α` and `β` are out_params, so this instance should not be dangerous
--- Porting note: @[nolint dangerous_instance]
+@[nolint dangerousInstance]
```

Comment:
This `nolint` is actually unnecessary now.  The new linter is much smarter than the old one (it understands out-params, and I've also taught it about the not-quite-out-params in CoeHead/CoeTail/Coe).

Reply:
This rule might be a bit on the cautious side, as it also rules out `Coe (Fin n) Nat` :latin_cross:, which is in core.  I'm open for other clear rules that rein in the coercion instances as well.  But coercion instances are an easy way to break everything (including notations and macros, since they rely on coercions!).  So we definitely need some strict linters here.

:latin_cross: (which notably conflicts with a hypothetical `Coe (Fin n) (Fin (n+1))`)

Reply:
I'm inclined to follow the cautious rule until we find a compelling counterexample to examine and derive some rules from. (The `Fin n -> Nat` coe can be made into a `CoeHead` as well, I think.)

Reply:
Yes, it can and probably should.

Reply:
Hmm, this breaks somewhat easily because we don't chain `CoeHead`.